src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,162 @@
+package duke;
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Duke {
+
+    protected int count = 0;
+    ArrayList<Task> list = new ArrayList<>();
+    protected Scanner in;
+
+    public void writeToFile(String filePath) throws IOException {
+        FileWriter file = new FileWriter(filePath);
+        for (Task task : list) {
+            file.write(task.fileFormat() + System.lineSeparator());
+        }
+        file.close();
+    }
+
+    public static void printAdded(Task task, int count) {
+        System.out.println("Got it. I've added this task:");
+        task.printTask();
+        System.out.println("Now you have " + count + " tasks in the list.");
+    }
+
+    public int executeCommand(String input) throws DukeException{
+        String command;
+        String description = null;
+        String time = null;
+        String temp;
+        if (input.contains(" ")) {
+            command = input.substring(0, input.indexOf(" "));
+            temp = input.substring(input.indexOf(" ") + 1);
+            if (temp.contains("/")) {
+                description = input.substring(input.indexOf(" ") + 1, input.indexOf("/") - 1);
+                time = input.substring(input.indexOf("/") + 4);
+            } else {
+                description = temp;
+            }
+        } else {
+            command = input;
+        }
```

Comment:
This can be extracted out to a separate method.
Think what you can do to improve SLAP

Reply:
Tried my best but some methods feels redundant especially if it contains only 1 line